File: Excel/476 Challenge.R
library(tidyverse)
library(readxl)

input1 = read_excel("Excel/476 Assigning Sales.xlsx", range = "A2:B5")
input2 = read_excel("Excel/476 Assigning Sales.xlsx", range = "D2:E11")
test   = read_excel("Excel/476 Assigning Sales.xlsx", range = "G2:I11")

result = input1 %>%
  left_join(input2, by = "Store") %>%
  mutate(n = n(), .by = Store) %>%
  mutate(Sales = Sales / n) %>%
  select(Store, Branch, Sales)

identical(result, test)
# [1] TRUE

File: Excel/400-499/476/476 Challenge.py
import pandas as pd

input1 = pd.read_excel("476 Assigning Sales.xlsx", usecols="A:B", skiprows=1, nrows = 3 )
input2 = pd.read_excel("476 Assigning Sales.xlsx", usecols="D:E", skiprows=1) 
input2.columns = ["Store", "Branch"]
test = pd.read_excel("476 Assigning Sales.xlsx", usecols="G:I", skiprows=1)
test.columns = ["Store", "Branch", "Sales"]

result = pd.merge(input1, input2, on="Store", how="left")
result["n"] = result.groupby("Store")["Store"].transform("count")
result["Sales"] = result["Sales"] / result["n"]
result["Sales"] = result["Sales"].astype('int64')
result = result[["Store", "Branch", "Sales"]]

print(result.equals(test)) # True